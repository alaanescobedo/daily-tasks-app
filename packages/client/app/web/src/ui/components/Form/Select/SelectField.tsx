import { forwardRef } from 'react'
import { InputGroup } from '../Input'
import { SelectBase } from './BaseSelect/SelectBase'

export const SelectField = forwardRef(({
  id,
  name,
  error,
  hint,
  label,
  required,
  size,
  gap,
  align,
  hideErrorField = false,
  keepHint = false,
  labelPosition,
  vertical,
  isInline,
  ...props
}: any, ref): JSX.Element => {
  const fieldProps = {
    id,
    name,
    error,
    hint,
    label,
    required,
    size
  }

  const stackProps = {
    gap,
    align,
    vertical
  }

  const baseProps = {
    isInline,
    size,
    ...props
  }

  return (
    <InputGroup
      labelPosition={labelPosition}
      hideErrorField={hideErrorField}
      keepHint={keepHint}
      {...fieldProps}
      {...stackProps}
    >
      <SelectBase {...baseProps} ref={ref} />
    </InputGroup>
  )
}
)
