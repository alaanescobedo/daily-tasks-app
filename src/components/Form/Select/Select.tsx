import { ChangeEvent } from 'react'
import sharedStyles from '@components/Form/shared/styles.module.css'

export type PriorityId = 'low' | 'medium' | 'high'
export type WeekdaysId = 'monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday' | 'sunday'

export type WeekdaysLabel = 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday' | 'Friday' | 'Saturday' | 'Sunday' | 'Today' | 'Tomorrow'
export type PriorityLabel = 'Low' | 'Medium' | 'High'

export interface InputSelectProps {
  id: string
  options: Array<{ id: PriorityId | WeekdaysId, label: PriorityLabel | WeekdaysLabel }>
  handleChange: (e: ChangeEvent<HTMLSelectElement>) => void
  defaultValue: string
}

export const Select = ({ id, options, handleChange, defaultValue }: InputSelectProps): JSX.Element => {
  return (
    <select name={id} id={id} className={`${sharedStyles.input_select}`} onChange={handleChange} defaultValue={defaultValue}>
      {options.map((option, index) => (
        <option key={index} value={option.id}>{option.label}</option>
      ))}
    </select>
  )
}
