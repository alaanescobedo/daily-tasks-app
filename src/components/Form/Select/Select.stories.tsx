import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Select, InputSelectProps } from './Select'

const InputSelectStory: ComponentMeta<typeof Select> = {
  title: 'Inputs/Select',
  component: Select
}
export default InputSelectStory

const Template: ComponentStory<typeof Select> = (args) => <Select {...args} />

export const PrioritySelect = Template.bind({})
const prioritySelectArgs: InputSelectProps = {
  id: 'priority',
  defaultValue: 'medium',
  handleChange: () => { },
  options: [
    { id: 'low', label: 'Low' },
    { id: 'medium', label: 'Medium' },
    { id: 'high', label: 'High' }
  ]
}
PrioritySelect.args = prioritySelectArgs

export const DateSelect = Template.bind({})
const dateSelectArgs: InputSelectProps = {
  id: 'date',
  defaultValue: 'friday',
  handleChange: () => { },
  options: [
    { id: 'friday', label: 'Friday' },
    { id: 'saturday', label: 'Saturday' },
    { id: 'sunday', label: 'Sunday' },
    { id: 'monday', label: 'Monday' },
    { id: 'tuesday', label: 'Tuesday' },
    { id: 'wednesday', label: 'Wednesday' },
    { id: 'thursday', label: 'Thursday' }
  ]
}
DateSelect.args = dateSelectArgs
