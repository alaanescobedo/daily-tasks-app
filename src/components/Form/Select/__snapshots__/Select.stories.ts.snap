// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Form/Select Date Select 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <select
      className="input_select"
      defaultValue="friday"
      id="date"
      name="date"
      onChange={[Function]}
    >
      <option
        value="friday"
      >
        Friday
      </option>
      <option
        value="saturday"
      >
        Saturday
      </option>
      <option
        value="sunday"
      >
        Sunday
      </option>
      <option
        value="monday"
      >
        Monday
      </option>
      <option
        value="tuesday"
      >
        Tuesday
      </option>
      <option
        value="wednesday"
      >
        Wednesday
      </option>
      <option
        value="thursday"
      >
        Thursday
      </option>
    </select>
     
  </div>,
  " ",
]
`;

exports[`Storyshots Form/Select Priority Select 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <select
      className="input_select"
      defaultValue="medium"
      id="priority"
      name="priority"
      onChange={[Function]}
    >
      <option
        value="low"
      >
        Low
      </option>
      <option
        value="medium"
      >
        Medium
      </option>
      <option
        value="high"
      >
        High
      </option>
    </select>
     
  </div>,
  " ",
]
`;
